runtime/moonbase/src/precompiles.rs

```diff
@@ -38,6 +40,26 @@ use sp_std::marker::PhantomData;
 #[derive(Debug, Clone, Copy)]
 pub struct MoonbasePrecompiles<R>(PhantomData<R>);
 
+impl<R> AccountIdToAssetId<R::AccountId, R::AssetId> for MoonbasePrecompiles<R>
+where
+	R: pallet_assets::Config,
+	R::AccountId: Into<H160>,
+	R::AssetId: From<u128>,
+{
+	fn account_to_asset_id(account: R::AccountId) -> Option<R::AssetId> {
+		let h160_account: H160 = account.into();
+		let mut data = [0u8; 16];
+		let (prefix_part, id_part) = h160_account.as_fixed_bytes().split_at(4);
+		if prefix_part == &[255u8; 4] {
+			data.copy_from_slice(id_part);
+			let asset_id: R::AssetId = u128::from_be_bytes(data).into();
+			Some(asset_id)
+		} else {
+			None
+		}
+	}
+}
+
```

Comment:
Every runtime that wants to use your precompile will need to do this with the current trait bounds. It's likely that there will be a few common ways to do it that can be re-used.

Reply:
Please check out the new layout of the bounds